tools/StaticAnalysis/ExampleAnalyzer/utils.ps1

```diff
@@ -68,138 +43,185 @@ class AnalysisOutput{
+        elseif($needRecord){
+            if($exampleLine.StartsWith("``````powershell","CurrentCultureIgnoreCase")){
+                # Find the head of code block with tag: powershell
+                $codeBlockComplete = $false
             }
-        }
-        else {
-            # From the start to the start of the first codeblock is example description.
-            $description = $exampleContent.SubString(0, $exampleCodeBlocks[0].Index).Trim()
-            if ($description -ne "") {
-                $exampleDescriptions += $description
+            elseif($exampleLine.StartsWith("``````output","CurrentCultureIgnoreCase")){
+                # Find the head of output block with tag: output
+                $outputBlockComplete = $false
+                $hasOutputBlock = $true
             }
-            # Extract code from the first "\n" to the last "\n"
-            foreach ($exampleCodeBlock in $exampleCodeBlocks) {
-                $code = $exampleCodeBlock.Value.Substring($exampleCodeBlock.Value.IndexOf("`n"), $exampleCodeBlock.Value.LastIndexOf("`n") - $exampleCodeBlock.Value.IndexOf("`n"))
-                if ($code -ne ""-and $code -notmatch "{{ Add code here }}") {
-                    $exampleCodes += $code
+            elseif($exampleLine -match "^``````\w+"){
+                $blockType = ($exampleLine -split "``````")[1]
+                if($blockType -notin $blockTypeList){
+                    $RuleName = "BlockTypeUnsupported"
+                    $ProblemID = 5062
+                    $Description = "The type of the tag after prompts `````` cannot be recognized."
+                    $Remediation = "Please check the type of the tag for this block."
+                    $errors += Set-AnalysisOutput $Module $Cmdlet $exampleNumber 0 $RuleName $ProblemID $missingSeverity $Description $Extent $Remediation
                 }
+                $otherTypeBlockComplete = $false
             }
-            # Extract output from the first "\n" to the last "\n"
-            foreach ($exampleOutputBlock in $exampleOutputBlocks) {
-                $output = $exampleOutputBlock.Value.Substring($exampleOutputBlock.Value.IndexOf("`n"), $exampleOutputBlock.Value.LastIndexOf("`n") - $exampleOutputBlock.Value.IndexOf("`n")).Trim()
-                if ($output -ne "") {
-                    $exampleOutputs += $output
+            elseif($exampleLine -match "\A\s*``````\s*\Z"){
+                if($codeBlockComplete -and $outputBlockComplete -and $otherTypeBlockComplete){
+                    if($firstCodeBlock){
+                        # Find the first head of code block without tag: powershell
+                        $codeBlockComplete = $false
+                        $firstCodeBlock = $false
+                    }
+                    else{
+                        $RuleName = "UnclearBlockType"
+                        $ProblemID = 5063
+                        $Description = "Too many blocks without declared type."
```

Comment:
Though we allow one single code block without language identifier, it's better to ask developers to always add language identifiers.
```suggestion
                        $Description = "The code blocks are missing language identifiers."
```